### PR TITLE
[Part 2 of 2] Constants loader and Config loader

### PR DESCRIPTION
**Pull request by martyall**
Finish up work started in #16186 . We probably want to merge these 2 at the same time or in close succession 

## The Problem
PR #16186 introduced a standard way for loading constants and config. In order to keep the diff to a minimum, I only ported the code contained in `src/lib` and `mina_cli_entrypoint`. The rest of the codebase still used the old way.

## The solution
This converts the rest of the executables to use the new `Constants` and `Config` loader.

### CODE REVIEW

**Comment by martyall** [comment]
!ci-nightly-me

**Comment by martyall** [comment]
!ci-nightly-me

**Comment by martyall** [comment]
nightly results (all important tests passing)
https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies/builds/2666

**Comment by martyall** [comment]
https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies/builds/2667

File: src/app/test_executive/test_executive.ml [hunk 1]
```diff
@@ -30,6 +30,7 @@ type inputs =
   ; mina_image : string
   ; archive_image : string option
   ; debug : bool
+  ; config_file : string list
```
Comment (georgeee) [inline src/app/test_executive/test_executive.ml:33]:
```suggestion
  ; config_files : string list
```

**Review by georgeee — APPROVED** [review]

**Comment by martyall** [comment]
!ci-nightly-me

**Comment by martyall** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me

**Review by joaosreis — APPROVED** [review]

**Review by mrmr1993 — CHANGES_REQUESTED** [review]
Don't change the CLI for all of the tools in a single PR. These are not routine changes, and each previous PR in this chain has been a large headache with breakages.

**Pull request closed without merging**